handle_hosts.go

```diff
@@ -0,0 +1,181 @@
+
+	"github.com/zazab/hierr"
+)
+
+func handleHosts(
+	zabbix *Zabbix,
+	config *Config,
+	args map[string]interface{},
+) error {
+	var (
+		hostnames, _ = parseSearchQuery(args["<pattern>"].([]string))
+		//addHost, _    = args["--add"].(string)
+		removeHost, _ = args["--remove"].(string)
+		confirmation  = !args["--noconfirm"].(bool)
+
+		hostids = []string{}
+		table   = tabwriter.NewWriter(os.Stdout, 1, 4, 2, ' ', 0)
+	)
+
+	var (
+		err               error
+		hosts, foundHosts []Host
+		hostid            string
+		messages          string
+	)
+
+	uniqKeys := make(map[string]bool)
+
+	switch {
+	case removeHost != "":
+
+		foundHosts, err = searchHosts(zabbix, removeHost)
+		if err != nil {
+			return hierr.Errorf(
+				err,
+				"can't obtain zabbix hosts",
+			)
+		}
+		for len(foundHosts) == 0 {
+			return nil
+		}
+
+		for len(foundHosts) > 1 {
+			return fmt.Errorf(
+				"found more then one uniq host for hostname: %s",
+				removeHost,
+			)
+		}
+
+		hostid = foundHosts[0].ID
+		fmt.Fprintf(
+			table,
+			"%s\t%s",
+			foundHosts[0].ID,
+			foundHosts[0].Name,
+		)
+
+		fmt.Fprint(table, "\n")
+
+		err = table.Flush()
+		if err != nil {
+			return err
+		}
+
+		messages = "removing"
+
+		if confirmation {
+			if !confirmHost(messages, removeHost) {
+				return nil
+			}
+		}
+
+		var hostRemoveIds Hosts
+
+		err = withSpinner(
+			":: Requesting for removing host",
+			func() error {
+				payload := []string{hostid}
+				hostRemoveIds, err = zabbix.RemoveHosts(payload)
+				return err
+			},
+		)
+		if err != nil {
+			return hierr.Errorf(
+				err,
+				"can't remove zabbix host",
+			)
+		}
+
+		return nil
+
+	default:
+
+		for _, hostname := range hostnames {
+			foundHosts, err = searchHosts(zabbix, hostname)
+			if err != nil {
+				return hierr.Errorf(
+					err,
+					"can't obtain zabbix hosts",
+				)
+			}
+			for _, host := range foundHosts {
+				if _, value := uniqKeys[host.ID]; !value {
+					uniqKeys[host.ID] = true
+					hosts = append(hosts, host)
+					hostids = append(hostids, host.ID)
+				}
+			}
+		}
+
+		for _, host := range hosts {
+
+			fmt.Fprintf(
+				table,
+				"%s\t%s",
+				host.ID,
+				host.Name,
+			)
+			fmt.Fprint(table, "\n")
+		}
+
+		err = table.Flush()
+		if err != nil {
+			return err
+		}
+
+	}
+
+	return nil
+}
+
+func confirmHost(messages, host string) bool {
+
+	var value string
+	fmt.Fprintf(
+		os.Stderr,
+		"\n:: Proceed with %s host %s? [Y/n]:",
+		messages,
+		host,
+	)
+
+	fmt.Scanln(&value)
+	return value == "" || value == "Y" || value == "y"
+}
+
+func searchHosts(zabbix *Zabbix, hostname string) ([]Host, error) {
+
+	var (
+		hosts []Host
+		err   error
+	)
+
+	params := Params{
+		"search": Params{
+			"name": hostname,
+		},
+		"output": []string{
+			"host",
+		},
+		"searchWildcardsEnabled": "1",
+	}
+
+	err = withSpinner(
+		":: Requesting information about hosts",
+		func() error {
+			hosts, err = zabbix.GetHosts(params)
+			return nil
+		},
+	)
+	if err != nil {
```

Comment:
Useless condition. You can do only one `return` statement here `return hosts, err`